corehq/apps/accounting/bootstrap/utils.py

```diff
@@ -6,58 +6,70 @@
 
 from corehq.apps.accounting.exceptions import AccountingError
 from corehq.apps.accounting.models import (
+    FeatureType,
     SoftwarePlanEdition,
     SoftwarePlanVisibility,
+    SoftwareProductType,
 )
 
 logger = logging.getLogger(__name__)
 
 
-def ensure_plans(edition_to_role, edition_to_product_rate, edition_to_feature_rate, feature_types, product_types,
-                 dry_run, verbose, apps):
+PRODUCT_TYPES = [
+    SoftwareProductType.COMMCARE,
+    SoftwareProductType.COMMCONNECT,
+    SoftwareProductType.COMMTRACK,
+]
+
+FEATURE_TYPES = [
+    FeatureType.USER,
+    FeatureType.SMS,
+]
+
+
+def ensure_plans(config, dry_run, verbose, apps):
     DefaultProductPlan = apps.get_model('accounting', 'DefaultProductPlan')
     SoftwarePlan = apps.get_model('accounting', 'SoftwarePlan')
     SoftwarePlanVersion = apps.get_model('accounting', 'SoftwarePlanVersion')
     Role = apps.get_model('django_prbac', 'Role')
 
-    editions = edition_to_role.keys()
-    edition_to_features = _ensure_features(feature_types, editions, dry_run=dry_run, verbose=verbose, apps=apps)
-    for product_type in product_types:
-        for edition in editions:
-            role_slug = edition_to_role[edition]
+    for product_type in PRODUCT_TYPES:
+        for plan_key, plan_deets in config.iteritems():
+            edition, is_trial = plan_key
+            features = _ensure_features(edition, dry_run, verbose, apps)
             try:
-                role = Role.objects.get(slug=role_slug)
+                role = _ensure_role(plan_deets['role'], apps)
             except Role.DoesNotExist:
-                logger.info("Could not find the role '%s'. Did you forget to run cchq_prbac_bootstrap?")
-                logger.info("Aborting. You should figure this out.")
                 return
```

Comment:
Was going to have this raise the exception, but on tests that would fail because `managed_hosting` and `reseller` roles aren't bootstrapped. It's not clear to me if they should be, but I think they would have to be added to the prbac bootstrap configs